### ISSUE TITLE
Node doesn't sync.

### ISSUE THREAD

**Issue by Myridium** [algorand/go-algorand#1190]
### Subject of the issue
The node doesn't sync. Similar issue to #571, probably.

### Your environment
* Software version: 
```
8590010404
2.0.75812.dev, commit #eeedd36d6
```
* Node status if applicable:
```
Last committed block: 0
Time since last block: 0.0s
Sync Time: 0.0s
Last consensus protocol: https://github.com/algorandfoundation/specs/tree/5615adc36bad610c7f165fa2967f4ecfa75125f0
Next consensus protocol: https://github.com/algorandfoundation/specs/tree/5615adc36bad610c7f165fa2967f4ecfa75125f0
Round for next consensus protocol: 1
Next consensus protocol supported: true
Last Catchpoint: 
Genesis ID: mainnet-v1.0
Genesis hash: wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=
```

* Operating System details. 
Arch Linux x64

* Last part of `node.log`:
```
{"Context":"Agreement","Hash":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA","ObjectPeriod":0,"ObjectRound":0,"ObjectStep":0,"Period":0,"Round":1,"Sender":"","Step":9,"Type":"VoteAttest","Weight":0,"WeightTotal":0,"file":"actions.go","function":"github.com/algorand/go-algorand/agreement.pseudonodeAction.do","level":"info","line":372,"msg":"attested to {{} 0 AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAY5HFKQ AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA} at (1, 0, 9)","time":"2020-06-21T22:40:54.165248+10:00"}
{"file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).meshThread","level":"info","line":1321,"msg":"got no DNS addrs for network mainnet","name":"","time":"2020-06-21T22:41:14.763916+10:00"}
{"Context":"Agreement","Hash":"","ObjectPeriod":0,"ObjectRound":0,"ObjectStep":0,"Period":0,"Round":1,"Sender":"","Step":9,"Type":"StepTimeout","Weight":0,"WeightTotal":0,"file":"trace.go","function":"github.com/algorand/go-algorand/agreement.(*tracer).logTimeout","level":"info","line":191,"msg":"timeout fired on (1, 0, 9) with value 4m31s (napping: false)","time":"2020-06-21T22:41:41.794256+10:00"}
{"file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).meshThread","level":"info","line":1321,"msg":"got no DNS addrs for network mainnet","name":"","time":"2020-06-21T22:42:14.763112+10:00"}
{"file":"logger.go","function":"github.com/algorand/go-algorand/daemon/algod/api/server/lib/middlewares.(*LoggerMiddleware).handler.func1","level":"info","line":56,"msg":"127.0.0.1:33724 - - [2020-06-21 22:42:36.481857307 +1000 AEST m=+325.785440615] \"GET /v2/status HTTP/1.1\" 200 524 \"Go-http-client/1.1\" 486.868µs","time":"2020-06-21T22:42:36.482399+10:00"}
{"file":"logger.go","function":"github.com/algorand/go-algorand/daemon/algod/api/server/lib/middlewares.(*LoggerMiddleware).handler.func1","level":"info","line":56,"msg":"127.0.0.1:33724 - - [2020-06-21 22:42:36.483498983 +1000 AEST m=+325.787082291] \"GET /versions HTTP/1.1\" 200 0 \"Go-http-client/1.1\" 123.606µs","time":"2020-06-21T22:42:36.483677+10:00"}
{"file":"logger.go","function":"github.com/algorand/go-algorand/daemon/algod/api/server/lib/middlewares.(*LoggerMiddleware).handler.func1","level":"info","line":56,"msg":"127.0.0.1:33726 - - [2020-06-21 22:42:39.883828253 +1000 AEST m=+329.187411561] \"GET /v2/status HTTP/1.1\" 200 524 \"Go-http-client/1.1\" 370.313µs","time":"2020-06-21T22:42:39.884253+10:00"}
{"file":"logger.go","function":"github.com/algorand/go-algorand/daemon/algod/api/server/lib/middlewares.(*LoggerMiddleware).handler.func1","level":"info","line":56,"msg":"127.0.0.1:33726 - - [2020-06-21 22:42:39.885528912 +1000 AEST m=+329.189112216] \"GET /versions HTTP/1.1\" 200 0 \"Go-http-client/1.1\" 164.393µs","time":"2020-06-21T22:42:39.885751+10:00"}
```

### Steps to reproduce
Install `go-algorand` as per instructions. Copy `go-algorand/installer/genesis/mainnet/genesis.json` to `~/.algorand/` and run `goal node start -d ~/.algorand`.

### Expected behaviour
Node should sync blockchain.

### Actual behaviour
Node does not sync blockchain. 

May be related to this message in the log: `got no DNS addrs for network mainnet`.

### Extra info
* Output of `nslookup -q=srv _algobootstrap._tcp.mainnet.algorand.network` appears normal; no 'authoritative answers' found.
* Output of `nslookup -q=srv _algobootstrap._tcp.mainnet.algorand.network 8.8.8.8` is `;; connection timed out; no servers could be reached`

**Comment by brianolson** [comment]
Based on `nslookup`, sounds like all DNS queries are failing. Is there a firewall rule preventing it? (Maybe something blocking outbound or inbound UDP?)

**Comment by Myridium** [comment]
How would I test this, @brianolson ?

**Comment by Myridium** [comment]
Okay, I connected to a VPN and it seems to be syncing now. Why was this necessary exactly...? 

Can we have `goal node status` actually report a very important aspect of the status of the node; namely that it cannot access the network? Very silly that it pretends everything is okay and just waiting.

**Comment by ian-algorand** [comment]
@tsachiherman, could you talk about our plans for addressing this for visibility here? Thanks!

**Comment by tsachiherman** [comment]
@Myridium , we have plans to extend the DNS setting to allow custom DNS servers, which I suspect would resolve the issue you see. Watch for changes in this issue to determine which release is going to receive these updates.

**Comment by algorandskiy** [comment]
Implemented in #1736. Now DNSSEC resolver attempts to use system resolver(s) first, then `FallbackDNSResolverAddress` from algod's config, and then defaults to hard-coded list of DNS servers.
